### ISSUE TITLE
Applying fixes with object syntax via `require-param` rule

### ISSUE THREAD

**Issue by seahindeniz** [gajus/eslint-plugin-jsdoc#869]
Hi,
Is there any setting/option to make the `require-param`  rule to apply fixes by adding an object instead of properties in new lines?
If not, I think moving forward with a new proposal should be acceptable 🙏🏻 

## Motivation

Applying fixes with object syntax in `@param` declarations.

## Current behavior

```js
/**
 * @param root0
 * @param root0.bar
 * @returns {void}
 */
const foo = ({ bar }) => {};
```

## Desired behavior

```js
/**
 * @param {{ bar: any }} param0
 * @returns {void}
 */
const foo = ({ bar }) => {};
```

## Alternatives considered

If it is worth to mention, copy-pasting types from TypeScript files with object syntax, makes it easier to adapt it into JSDoc.

**Comment by brettz9** [comment]
Sounds like a good option to add. Note that although their logic may be more complicated, there are a couple bugs on the param checking rules (#540 and #762 ) that we can hopefully get fixed.
